app/src/main/java/ru/ifmo/md/lesson8/AddCityActivity.java

```diff
@@ -0,0 +1,90 @@
+package ru.ifmo.md.lesson8;
+
+import android.content.ContentValues;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import ru.ifmo.md.lesson8.db.CitiesTable;
+import ru.ifmo.md.lesson8.db.WeatherContentProvider;
+import ru.ifmo.md.lesson8.service.AsyncResponseReceiver;
+import ru.ifmo.md.lesson8.service.CitySearchTask;
+import ru.ifmo.md.lesson8.service.ForecastParser;
+import ru.ifmo.md.lesson8.service.ForecastService;
+import ru.ifmo.md.lesson8.service.Throttler;
+
+
+public class AddCityActivity extends ActionBarActivity implements TextWatcher, AsyncResponseReceiver<ForecastParser.LocationResult> {
+
+    EditText searchField;
+    TextView suggestion;
+    ForecastParser.LocationResult currentResult;
+    Throttler mThrottler = new Throttler();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_city);
+
+        searchField = (EditText) findViewById(R.id.search_field);
+        suggestion = (TextView) findViewById(R.id.suggestion);
+
+        searchField.addTextChangedListener(this);
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        if(s.length() > 0) {
+            final String name = s.toString();
+            final CitySearchTask task = new CitySearchTask(this);
+            Throttler.Callback foo = new Throttler.Callback() {
+                @Override
+                public void call() {
+                    task.execute(name);
+                }
+            };
+            mThrottler.throttle(foo, 500);
```

Comment:
The idea is not to send a request until user stops typing. Throttler waits given number of ms before starting the task, and this timeout is resetted if `throttle()` is called again before expiration. As a result, nothing happens when text in the field changes frequently and a request is sent only half a second after text stopped changing.